contracts/base/strategies/mesh/MeshStakingStrategyBase.sol

```diff
@@ -0,0 +1,138 @@
+  /// @notice Initialize contract after setup it as proxy implementation
+  function initializeStrategy(
+    address _controller,
+    address _underlying,
+    address _vault,
+    address[] memory __rewardTokens
+  ) public initializer {
+    ProxyStrategyBase.initializeStrategyBase(
+      _controller,
+      _underlying,
+      _vault,
+      __rewardTokens,
+      _BUY_BACK_RATIO
+    );
+  }
+
+  event VotingAdded(address exchange, uint256 amount);
+  event VotingRemoved(address exchange, uint256 amount);
+
+  // ------------------ GOV actions --------------------------
+
+  /// @dev Manual withdraw for any emergency purposes
+  function manualWithdraw() external restricted {
+    IERC20(_underlying()).safeTransfer(_vault(), IERC20(_underlying()).balanceOf(address(this)));
+  }
+
+  function addVoting(address exchange, uint256 amount) external restricted {
+    require(exchange != address(0), "Exchange address should be specified");
+    poolVoting.addVoting(exchange, amount);
+    emit VotingAdded(exchange, amount);
+  }
+
+  function removeVoting(address exchange, uint256 amount) external restricted {
+    require(exchange != address(0), "Exchange address should be specified");
+    poolVoting.removeVoting(exchange, amount);
+    emit VotingRemoved(exchange, amount);
+  }
+
+  // --------------------------------------------
+
+  /// @notice Return only pool balance. Assume that we ALWAYS invest on vault deposit action
+  function investedUnderlyingBalance() external override view returns (uint) {
+    return _rewardPoolBalance();
+  }
+
+  /// @dev Returns MESH amount under control
+  function _rewardPoolBalance() internal override view returns (uint256) {
+    return vMesh.lockedMESH(address(this));
+  }
+
+  /// @dev In this version rewards are accumulated in this strategy
+  function doHardWork() external onlyNotPausedInvesting override hardWorkers {
+    vMesh.claimReward();
+  }
+
+  /// @dev Stake Mesh to vMesh
+  function depositToPool(uint256 amount) internal override {
+    if (amount > _MESH_PRECISION) {
+      // lock on max period
+      // mesh allows only integer values w/o precision e.g 1 mesh token
+      amount = amount / _MESH_PRECISION; // rounding
+      IERC20(_underlying()).safeApprove(address(vMesh), 0);
+      IERC20(_underlying()).safeApprove(address(vMesh), amount * _MESH_PRECISION);
+      vMesh.lockMESH(amount, _MAX_LOCK_PERIOD);
+    }
+  }
+
+  /// @dev Not supported by MESH
+  function withdrawAndClaimFromPool(uint256) internal pure override {
+    revert("MSS: Withdraw forbidden");
+  }
+
+  /// @dev Not supported by MESH
+  function emergencyWithdrawFromPool() internal pure override {
+    revert("MSS: Method not supported");
+  }
+
+  /// @dev In this version MESH rewards are not liquidated.
+  function liquidateReward() internal override {}
+
+  /// @dev No claimable tokens
+  function readyToClaim() external view override returns (uint256[] memory) {
```

Comment:
you can name output variable and do not redeclare it
```
returns (uint256[] memory toClaim ) {
  toClaim = new uint256[](_rewardTokens.length);
}
```

Reply:
Done